spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -44,19 +44,20 @@ def get_parser():
         help='Binary mask of lesions (lesions are labeled as "1").',
         metavar=Metavar.file,
     )
-    mandatory_arguments.add_argument(
-        "-s",
-        required=True,
-        help="Spinal cord centerline or segmentation file, which will be used to correct morphometric measures with "
-             "cord angle with respect to slice. (e.g.'t2_seg.nii.gz')",
-        metavar=Metavar.file)
 
     optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
     optional.add_argument(
         "-h",
         "--help",
         action="help",
         help="show this help message and exit")
+    optional.add_argument(
+        "-s",
+        required=False,
+        help="Spinal cord centerline or segmentation file, which will be used to correct morphometric measures with "
+             "cord angle with respect to slice. (e.g.'t2_seg.nii.gz')\nIf not provided, only lesion volume will be "
+             "computed.",
```

Comment:
Small nitpick, but I think we can be a little clearer about what will and won't be computed:

```suggestion
        help="Spinal cord centerline or segmentation file, which will be used to correct morphometric measures with "
             "cord angle with respect to slice. (e.g. 't2_seg.nii.gz')\n"
             "If provided, then the lesion volume, length, and diameter will be computed. Otherwise, if not provided, "
             "then only the lesion volume will be computed.",
```